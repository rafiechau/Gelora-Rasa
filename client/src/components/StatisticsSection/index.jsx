// StatisticsSection.jsx
import React from 'react';
import { FormattedMessage } from 'react-intl';
import classes from './style.module.scss';

const StatisticsSection = () => (
  <div className={classes.statistics} data-testid="statistics-section">
    <div className={classes.statistic}>
      <span className={classes.statNumber}>150+</span>
      <span className={classes.statLabel}>
        <FormattedMessage id="app_events_attended" />
      </span>
    </div>
    <div className={classes.statistic}>
      <span className={classes.statNumber}>3000+</span>
      <span className={classes.statLabel}>
        <FormattedMessage id="app_events_tickets_sold" />
      </span>
    </div>
  </div>
);

export default StatisticsSection;
