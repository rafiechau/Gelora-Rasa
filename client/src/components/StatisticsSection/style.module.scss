.statistics {
    display: flex;
    justify-content: space-around;
    flex-wrap: wrap;
    margin: 2rem 0;
    padding: 1rem;
    background-color: lighten($navy, 5%); // A slightly lighter navy shade for contrast
    border-radius: 10px; // Consistent with the card styling
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1); // Subtle shadow for depth
    @media (max-width: 768px) {
      flex-direction: row;
      justify-content: space-between;
    }
    @media (max-width: 480px) {
      flex-direction: column;
      align-items: center;
    }
    .statistic {
      display: flex;
      flex-direction: column;
      align-items: center;
      padding: 1rem;
      margin: 1rem;
      background-color: $navy; 
      border-radius: 10px; 
      width: 200px; 
      box-shadow: inset 0 2px 4px rgba(0, 0, 0, 0.3); 
      @media (max-width: 768px) {
        width: 30%;
      }
      @media (max-width: 480px) {
        width: 80%;
      }
  
      .statNumber {
        font-size: 2rem; 
        font-weight: bold;
        color: $color-white;
        margin-bottom: 0.5rem;
      }
  
      .statLabel {
        font-size: 0.9rem;
        color: $color-white-fade;
        text-transform: uppercase;
      }
    }
  }
  