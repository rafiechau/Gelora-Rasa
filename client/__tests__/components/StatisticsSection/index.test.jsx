import React from 'react';
import { render, screen } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import StatisticsSection from '@components/StatisticsSection';

describe('StatisticsSection', () => {
  beforeEach(() => {
    render(
      <IntlProvider
        locale="en"
        messages={{
          app_events_attended: 'Events Attended',
          app_events_tickets_sold: 'Tickets Sold',
        }}
      >
        <StatisticsSection />
      </IntlProvider>
    );
  });

  test('renders the statistics section', () => {
    expect(screen.getByTestId('statistics-section')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const statisticsSection = screen.getByTestId('statistics-section');
    expect(statisticsSection).toMatchSnapshot();
  });
});
