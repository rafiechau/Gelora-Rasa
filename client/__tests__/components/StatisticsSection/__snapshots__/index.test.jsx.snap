// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticsSection matches snapshot 1`] = `
<div
  class="statistics"
  data-testid="statistics-section"
>
  <div
    class="statistic"
  >
    <span
      class="statNumber"
    >
      150+
    </span>
    <span
      class="statLabel"
    >
      Events Attended
    </span>
  </div>
  <div
    class="statistic"
  >
    <span
      class="statNumber"
    >
      3000+
    </span>
    <span
      class="statLabel"
    >
      Tickets Sold
    </span>
  </div>
</div>
`;
